src/librustdoc/clean/types.rs

```diff
@@ -1057,14 +1034,99 @@ pub(crate) fn hir_attr_lists<'a, I: IntoIterator<Item = &'a hir::Attribute>>(
         .flatten()
 }
 
+/// This type keeps track of (doc) cfg information as we go down the item tree.
```

Comment:
(feel free to ignore) I wonder if moving all of these cfg-related things below into the "propagate cfg" module, so `CfgInfo`, `extract_cfg_from_attrs` etc. Tho it's kinda weird to have other parts of rustdoc import from `passes/`... idk